engine/src/main/java/org/pentaho/di/repository/RepositoryConnectionUtils.java

```diff
@@ -0,0 +1,178 @@
+      } else {
+        // TODO: add support for specified repositories.xml files...
+        repositoriesMeta.readData(); // Read from the default $HOME/.kettle/repositories.xml file.
+      }
+    } catch ( Exception e ) {
+      throw new ActionExecutionException( BaseMessages.getString(
+        "RepositoryConnectionUtils.ERROR_0018_META_REPOSITORY_NOT_POPULATED" ), e ); //$NON-NLS-1$
+    }
+
+    if ( logger.isDebugEnabled() ) {
+      logger.debug( " Finding repository metadata" );
+    }
+    // Find the specified repository.
+    RepositoryMeta repositoryMeta = null;
+    try {
+      if ( isSingleDiServerInstance ) {
+        repositoryMeta = repositoriesMeta.findRepository( SINGLE_DI_SERVER_INSTANCE );
+      } else {
+        repositoryMeta = repositoriesMeta.findRepository( repositoryName );
+      }
+
+
+    } catch ( Exception e ) {
+      throw new ActionExecutionException( BaseMessages.getString( PKG,
+        "RepositoryConnectionUtils.ERROR_0004_REPOSITORY_NOT_FOUND", repositoryName ), e ); // $NON-NLS-1$
+    }
+
+    if ( repositoryMeta == null ) {
+      if ( logger.isDebugEnabled() && logBuffer != null ) {
+        logger.debug( logBuffer.getBuffer().toString() );
+      }
+      throw new ActionExecutionException( BaseMessages.getString( PKG,
+        "RepositoryConnectionUtils.ERROR_0004_REPOSITORY_NOT_FOUND", repositoryName ) ); //$NON-NLS-1$
+    }
+
+    if ( logger.isDebugEnabled() ) {
+      logger.debug( " Getting repository instance " ); //$NON-NLS-1$
+    }
+    Repository repository = null;
+    try {
+      repository =
+        PluginRegistry.getInstance().loadClass( RepositoryPluginType.class, repositoryMeta.getId(),
+          Repository.class );
+      repository.init( repositoryMeta );
+
+    } catch ( Exception e ) {
+      throw new ActionExecutionException( BaseMessages.getString(
+        "RepositoryConnectionUtils.ERROR_0016_COULD_NOT_GET_REPOSITORY_INSTANCE" ), e ); // $NON-NLS-1$
+    }
+
+    // OK, now try the username and password
+    if ( logger.isDebugEnabled() ) {
+      logger.debug( "Connecting to repository " );
+    }
+
+    // Two scenarios here: internal to server or external to server. If internal, you are already authenticated. If
+    // external, you must provide a username and additionally specify that the IP address of the machine running this
+    // code is trusted.
+    repository.connect( userName, "ignore" );
+
+    // OK, the repository is open and ready to use.
+    if ( logger.isDebugEnabled() ) {
+      logger.debug( " Connected to repository " );
+    }
+
+    return repository;
+  }
+
+  private static String getEmbeddedDefaultRepositoryXml( final String fullyQualifiedServerUrl ) {
+    // only load a default enterprise repository. If this option is set, then you cannot load
+    // transformations or jobs from anywhere but the local server.
+
+    String repositoriesXml = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><repositories>" //$NON-NLS-1$
```

Comment:
**Intellij warning:** Initializing this variable is redundant, the same result can be done with:
*return "<?xml version=\"1.0\" encoding=\"UTF-8\"?><repositories>" ...*

Reply:
Fixed